### PR TITLE
Improve readability

### PR DESCRIPTION
**Pull request by rhaschke**
This PR proposes:
1. More readable hand types:
   - hand_e -> motor
   - hand_e_plus -> motor+
   - hand_m -> muscle
   - hand_mt -> muscletrans
2. A simplified `fingers` specification:
   - `fingers=all` (for a fully-fingered hand)
   - `fingers='th, mf, rf'` (textual list, acceptable separaters could be spaces, commas, and semicolons)
3. Support all types of tactile sensor specifications:
     - [x] plaing string: `tip_sensors=bio` (same sensor for all fingers)
     - [x] list-like: `tip_sensors='bio, bio, std, std, none'` (assuming order th ff mf rf lf)
     - [x] dict-like: `tip_sensors='th=bio, ff=bio, mf=std, rf=std, lf=none'`

See individual commit messages for details.
This PR requires the latest `xacro` melodic or noetic branch from [our fork](https://github.com/ubi-agni/xacro). 
I will trigger a release asap - if there are no issues running our xacro code.

### TODO
- [ ] Validate xacro extension on our xacro code base
- [x] Fix corner cases failing when `running test/test_compatibility.py`

### CODE REVIEW

**Comment by rhaschke** [comment]
@guihomework @beatrizleon
From my point of view, this is ready for review.

File: sr_description/hand/xacro/three_biotac_fingers_hand.urdf.xacro [hunk 1]
```diff
@@ -10,11 +10,10 @@
...
   params="muscletrans muscle bio bt_sp ubi eli reflect prefix">
     ${xacro.warning("The file three_biotac_fingers_hand.urdf.xacro is deprecated", print_location=False)}
-    ${xacro.warning("Use full_hand.urdf.xacro instead with tip_sensors=dict([(f, 'bio') for f in 'th ff mf'.split()])")}
+    ${xacro.warning("Use full_hand.urdf.xacro instead with tip_sensors='th=bio ff=bio mf=bio'")}
     <!-- forward to generic macro -->
     <xacro:include filename="$(find sr_description)/hand/xacro/full_hand.urdf.xacro" />
     <!-- BioTacs on First and Middle Finger as well as Thumb -->
-    <xacro:shadowhand reflect="${reflect}" prefix="${prefix}"
-    tip_sensors="${['bio', 'bio', 'bio', 'std', 'std']}"/>
+    <xacro:shadowhand reflect="${reflect}" prefix="${prefix}" tip_sensors="'th=bio ff=bio mf=bio'" />
```
Comment (guihomework) [inline sr_description/hand/xacro/three_biotac_fingers_hand.urdf.xacro]:
```suggestion
    <xacro:shadowhand reflect="${reflect}" prefix="${prefix}" tip_sensors="th=bio ff=bio mf=bio" />
```
to be similar to sr_description/robots/shadowhand_motor_th_ff_rf_ellipsoid.urdf.xacro which also does not need the extra single quotes. 
Careful, this macro is never instantiated (and was never in the past either, so will not be tested by the compatibility test right now). I did a manual test with some hand crafted instantiation in old and new form, and the result is fine.

**Review by guihomework — APPROVED** [review]
all test pass, I see no issues except the tiny change in the comment

File: sr_description/hand/xacro/three_biotac_fingers_hand.urdf.xacro (hunk 1 above)
Comment (rhaschke) [reply sr_description/hand/xacro/three_biotac_fingers_hand.urdf.xacro]: Thanks. Applied.
